pennylane/hf/hamiltonian.py

```diff
@@ -240,52 +240,53 @@ def hamiltonian(*args):
         Returns:
             Hamiltonian: the qubit Hamiltonian
         """
-        h_ferm = generate_fermionic_hamiltonian(mol, cutoff, core, active)(*args)
-
-        ops = []
-
-        for n, t in enumerate(h_ferm[1]):
-
-            if len(t) == 0:
-                coeffs = anp.array([0.0])
-                coeffs = coeffs + np.array([h_ferm[0][n]])
-                ops = ops + [qml.Identity(0)]
-
-            elif len(t) == 2:
-                op = _generate_qubit_operator(t)
-                if op != 0:
-                    for i, o in enumerate(op[1]):
-                        if len(o) == 0:
-                            op[1][i] = qml.Identity(0)
-                        if len(o) == 1:
-                            op[1][i] = _return_pauli(o[0][1])(o[0][0])
-                        if len(o) > 1:
-                            k = qml.Identity(0)
-                            for o_ in o:
-                                k = k @ _return_pauli(o_[1])(o_[0])
-                            op[1][i] = k
-                    coeffs = np.concatenate([coeffs, np.array(op[0]) * h_ferm[0][n]])
-                    ops = ops + op[1]
-
-            elif len(t) == 4:
-                op = _generate_qubit_operator(t)
-                if op != 0:
-                    for i, o in enumerate(op[1]):
-                        if len(o) == 0:
-                            op[1][i] = qml.Identity(0)
-                        if len(o) == 1:
-                            op[1][i] = _return_pauli(o[0][1])(o[0][0])
-                        if len(o) > 1:
-                            k = qml.Identity(0)
-                            for o_ in o:
-                                k = k @ _return_pauli(o_[1])(o_[0])
-                            op[1][i] = k
-                    coeffs = np.concatenate([coeffs, np.array(op[0]) * h_ferm[0][n]])
-                    ops = ops + op[1]
-
-        h = simplify(qml.Hamiltonian(coeffs, ops), cutoff=cutoff)
-
-        return h
+        with qml.tape.stop_recording():  # prevent unowned identities from being queued
```

Comment:
Thanks @albi3ro. This function in `qml.hf` will be deprecated soon. I am curious if the problem also exists when using the new [mol_hamiltonian](https://github.com/PennyLaneAI/pennylane/blob/289db9d92d70d3c8f6ac75966980d472991276cc/qchem/pennylane_qchem/qchem/hf/hamiltonian.py#L181) function in `qml.qchem.hf`?

Note that the code you put inside the `with` statement is now converted to independent functions in the [observable](https://github.com/PennyLaneAI/pennylane/blob/master/qchem/pennylane_qchem/qchem/hf/observable.py) module.

We also have another observable, dipole moment, which is differentiable and may suffer from the same issue.

Reply:
Installing qchem now to try and find out.  Hopefully the dependencies cooperate with an M1 Mac.  I'm betting this context manager will have to be added in more places.

Reply:
@soranjh  Yes `mol_hamiltonian` was also queuing a bunch of identities.  Seems like there's a bunch of functions in `qml.qchem.hf` that are queuing extra identities.

Reply:
Do you know what functions are they? (other than `mol_hamiltonian` and `dipole_moment`)